ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/XAResourceRecord.java

```diff
@@ -918,6 +953,7 @@ public boolean restore_state(InputObjectState os, int t)
 	    {
 	        _heuristic = os.unpackInt();
 	        _committed = os.unpackBoolean();
+					_rolledBack = os.unpackBoolean();
```

Comment:
@jmfinelli would you agree to add a test that verifies this _rolledBack behaviour is working as expected after recovery, please? I can work on this instead if you prefer?

Reply:
I have created a test for review, https://github.com/jmfinelli/narayana/pull/3. Though I would mention that _rolledBack seems only useful to be saved for serializable XAResources

Reply:
I doubt many people will be using serializable XAResources so is thi PRs fixing a corner case or is there more to this issue than it seems?

Reply:
Most of the pull request is fixing the case where we are calling erroneously calling xa_rollback when a resource manager returned a rollback condition during prepare and that is beneficial to all users.

Persisting the value is logical and thus appropriate to include, but it does not seem to benefit all XAResource implementations and breaks compatibility with the object store format for this record type so that would need describing and considering for any release that does not imply starting with a fresh object store.

Reply:
We removed the braking change so this can be resolved. Thanks